### PR TITLE
added the mutual information calculation

### PR DESCRIPTION
**Pull request by wildart**

### CODE REVIEW

File: src/mutualinfo.jl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    hk = entAcols/N + log(N)
+
+    mi = hc - hck
+    nmi = 2*mi/(hc+hk)
+
+    return mi, nmi
+end
+
+"""
+    mutualinfo(a, b) -> Float64
```
Comment (alyst) [inline src/mutualinfo.jl]: Maybe we can add a keyword specifying whether we need the normalized MI?
```suggestion
    mutualinfo(a, b; normed::Bool = false) -> Float64
```

`multualinfo(a, b, normed=true)` is better than `mutualinfo(a, b)[2]`

File: src/mutualinfo.jl [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Mutual Information
+
+function _mutualinfo(A::AbstractMatrix{<:Integer})
+    N = sum(A)
+    (N == 0.0) && return 0.0
+
+    aᵢ = sum(A, dims=2)
```
Comment (alyst) [inline src/mutualinfo.jl]: That looks cool, but I'm not sure it's immediately obvious from the var names that they refer to the cluster sizes of `a` and `b` from `mutualinfo(a, b)`.
FWIW, I'd keep it as it was in `vmeasure`.

**Comment by wildart** [comment]
Should be fine now.

File: src/mutualinfo.jl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+"""
+    mutualinfo(a, b; normed=true) -> Float64
+
+Compute the *mutual information* between the two clusterings of the same
+data points.
+
+`a` and `b` can be either [`ClusteringResult`](@ref) instances or
+assignments vectors (`AbstractVector{<:Integer}`).
+
+If `normed` paremeter is `true`, the normalized mutual information (sum) is returned.
```
Comment (alyst) [inline src/mutualinfo.jl]:
```suggestion
If `normed` parameter is `true`, the normalized mutual information (sum) is returned.
```

Comment (alyst) [reply src/mutualinfo.jl]: I'm not sure I 100% understand "normalized mutual information (sum)" mean. What is "sum" in this context?

**Comment by alyst** [comment]
@wildart Thank you! Could you please fix a tiny typo and maybe clarify a bit the "normed = true" case, then I'll merge the PR.

**Pull request merged**